enterprise_access/apps/bffs/tests/test_response_builders.py

```diff
@@ -0,0 +1,143 @@
+import ddt
+from django.test import RequestFactory, TestCase
+
+from enterprise_access.apps.api_client.tests.test_license_manager_client import TestLicenseManagerUserApiClient
+from enterprise_access.apps.bffs.context import HandlerContext
+from enterprise_access.apps.bffs.response_builder import BaseLearnerResponseBuilder, BaseResponseBuilder
+from enterprise_access.apps.bffs.tests.test_handlers import TestBaseHandlerMixin
+
+
+@ddt.ddt
+class TestBaseResponseBuilder(TestBaseHandlerMixin):
+    def setUp(self):
+        super().setUp()
+        self.base_response_builder = BaseResponseBuilder(self.context)
+
+    def test_base_build_error(self):
+        with self.assertRaises(NotImplementedError):
+            self.base_response_builder.build()
+
+    @ddt.data(
+        {
+            'errors': True,
+            'warnings': True,
+        },
+        {
+            'errors': True,
+            'warnings': False,
+        },
+        {
+            'errors': False,
+            'warnings': True,
+        },
+        {
+            'errors': False,
+            'warnings': False,
+        }
+    )
+    @ddt.unpack
+    def test_add_errors_warnings_to_response(self, errors, warnings):
+        sample_response = {
+            "some_existing_metadata": "should_still_exist"
+        }
+        expected_output = {
+            **sample_response,
+        }
+        if errors:
+            self.context.add_error(
+                **self.mock_error
+            )
+            expected_output['errors'] = [self.mock_error]
+        if warnings:
+            self.context.add_warning(
+                **self.mock_warning
+            )
+            expected_output['warnings'] = [self.mock_warning]
+        response_data = self.base_response_builder.add_errors_warnings_to_response(sample_response)
+        self.assertEqual(response_data, expected_output)
+
+    @ddt.data(
+        {
+            'status_code': 201
```

Comment:
nit: I'm not sure the API will resolve to a `201`, even if it has side effects like auto-applying a subscription license; in this case, I think the status code would still be 200. Given this, perhaps use the test to reflect an error status code like `400` instead?

nit x2: consider using `status.HTTP_400_BAD_REQUEST`